### PR TITLE
[docs] P-tuning for sequence classification

### PR DESCRIPTION
**Pull request by stevhliu**
Moves the p-tuning for sequence classification [notebook](https://github.com/huggingface/peft/blob/main/examples/sequence_classification/P_Tuning.ipynb) into the docs.

### CODE REVIEW

File: docs/source/task_guides/ptuning-seq-classification.mdx [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    optimizer=optimizer,
+    num_warmup_steps=0,
+    num_training_steps=(len(train_dataloader) * num_epochs),
+)
+```
+
+Move the model to the GPU, and then write a training loop that includes the `metric` you loaded earlier to evaluate the model at each epoch:
+
+```py
+model.to(device)
```
Comment (MKhalusova) [inline docs/source/task_guides/ptuning-seq-classification.mdx:150]: Is there a reason why we don't use Trainer from Transformers here?

**Review by MKhalusova — COMMENTED** [review]
Thank you for creating this doc! I have left a couple of comments, but my main question is - why don't we use Trainer here?

File: docs/source/task_guides/ptuning-seq-classification.mdx (hunk 1 above)
Comment (pacman100) [reply docs/source/task_guides/ptuning-seq-classification.mdx:155]: Hello @stevhliu  and @MKhalusova, Trainer and Accelerate are closely supported by PEFT. 

All the example scripts use Accelerate, for example, https://github.com/huggingface/peft/blob/main/examples/conditional_generation/peft_lora_seq2seq_accelerate_ds_zero3_offload.py and https://github.com/huggingface/peft/blob/main/examples/causal_language_modeling/peft_lora_clm_accelerate_ds_zero3_offload.py. 

All the INT8 quantization examples mostly use Trainer, for example, https://github.com/huggingface/peft/blob/main/examples/int8_training/Finetune_flan_t5_large_bnb_peft.ipynb and https://github.com/huggingface/peft/blob/main/examples/int8_training/peft_bnb_whisper_large_v2_training.ipynb

Sequence Classification examples are simple and I habitually use native PyTorch in notebooks as I mostly write scripts with Accelerate which closely follows native PyTorch. Feel free to switch to Trainer/Accelerate notebook launcher for this example.

**Review by pacman100 — APPROVED** [review]
Thank you @stevhliu for adding the doc and @MKhalusova for suggesting the use of trainer. ✨

**Review by MKhalusova — APPROVED** [review]
Great work!
